### ISSUE TITLE
Error: Must have a country code to connect to a Tap to Pay reader on SDK version update to 3.0.0

### ISSUE THREAD

**Issue by ajithv11** [stripe/stripe-terminal-android#377]
Hi,

Latest SDK version update from 2.22.0 to 3.0.0 is resulting an error **Must have a country code to connect to a Tap to Pay reader** when I tried to connect local mobile reader, The changes I have added is given below

    /*val config = DiscoveryConfiguration(
    timeout = 0,
    discoveryMethod = DiscoveryMethod.LOCAL_MOBILE,
    isSimulated = true,
    location = Constants.LOCATIONID
    )*/ // This is changed for v3.0.0

    val config = DiscoveryConfiguration.LocalMobileDiscoveryConfiguration(
                  isSimulated = true
                 )
    discoveryTask = Terminal.getInstance().discoverReaders(config,
      
      discoveryListener = object : DiscoveryListener {
          override fun onUpdateDiscoveredReaders(readers: List<Reader>) {
          val firstReader = readers.first();
          this@TapViewModel.reader.postValue(firstReader);
          val config = ConnectionConfiguration.LocalMobileConnectionConfiguration(Constants.LOCATIONID);
          
          Terminal.getInstance().connectLocalMobileReader(firstReader, config,
          
          object : ReaderCallback {

                override fun onSuccess(reader: Reader) {

                }

                override fun onFailure(e: TerminalException) {
                    Log.d("Response Tap", "" + e.errorMessage);  // Must have a country code to connect to a Tap to Pay reader
                }

            });

          }
      },

      object : Callback {

        override fun onSuccess() {
            Log.d("Response Tap", "Readers Discovered Success!");
        }

        override fun onFailure(e: TerminalException) {
            Log.d("Response Tap", "Reader Discovery Failed : " + e.errorMessage);
        }

    })

**Terminal.getInstance().connectLocalMobileReader** goes to success for v2.22.0 , I couldn't figure out this issue

**Comment by chr-stripe** [comment]
This is a duplicate of #372 – it will be fixed in the upcoming version `3.1.0`. I will update this ticket when the next version is released.

**Comment by h0jeZvgoxFepBQ2C** [comment]
Any idea when this will be released?

**Comment by h0jeZvgoxFepBQ2C** [comment]
Right now there is no workaround as far as I understand it.. So we can't use 3.0.0 at all?

**Comment by grishmaukani-eb** [comment]
I am also facing the same issue. @h0jeZvgoxFepBQ2C @chr-stripe

**Comment by chr-stripe** [comment]
The `3.1.0` release will be available early next week.

**Comment by ajithv11** [comment]
@chr-stripe  Thanks for your reply!  

One quick question , Could you please tell me if I can use the SDK version 2.22.0  without upgrading also considering that in future does this older version of SDK will get obsolete?

**Comment by h0jeZvgoxFepBQ2C** [comment]
Also its not really cool that there is now only the documentation available for the broken version, the old documentation is not online anymore :(

**Comment by h0jeZvgoxFepBQ2C** [comment]
Also its not really cool that there is now only the documentation available for the broken version, the old documentation is not online anymore :(

**Comment by chr-stripe** [comment]
@ajithv11 yes, you can continue to use 2.x versions of the SDK while 3.X is actively being updated. Note that we will continue to occasionally backport critical fixes and security improvements to the latest 2.x release, if necessary.

We are currently planning to maintain support for 2.x SDKs through until mid-2024. You can view our timelines for SDK support in this repo's [SUPPORT.md](https://github.com/stripe/stripe-terminal-android/blob/master/SUPPORT.md) file.

**Comment by chr-stripe** [comment]
@h0jeZvgoxFepBQ2C documentation for the latest version of each major revision will always continue to be available on `stripe.dev`:
- [(Current) 3.x API Reference](https://stripe.dev/stripe-terminal-android)
- [2.x API Reference](https://stripe.dev/stripe-terminal-android/v2)
- [1.x API Reference](https://stripe.dev/stripe-terminal-android/v1)

**Comment by chr-stripe** [comment]
Hi folks, SDK `3.1.0` is now available and should include a fix for simulated Tap to Pay readers. Thank you for your patience!
